File: 0209-minimum-size-subarray-sum/209.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        left, running_sum = (0, 0)
        
        ans = n + 1
        for right in range(n):
            running_sum += nums[right]
            while running_sum >= target:
                ans = min(ans, right - left + 1)
                running_sum -= nums[left]
                left += 1

        return 0 if ans == n + 1 else ans

File: 0209-minimum-size-subarray-sum/209.rs
impl Solution {
    pub fn min_sub_array_len(target: i32, nums: Vec<i32>) -> i32 {
        let n = nums.len(); 
        let (mut left, mut running_sum) = (0, 0);

        let mut ans = n + 1;
        for right in 0..n {
            running_sum += nums[right];
            while running_sum >= target {
                ans = ans.min(right - left + 1);
                running_sum -= nums[left];
                left += 1;
            }
        }

        if ans == n + 1 {0} else {ans as i32}
    }
}
